File: proga/1_semestr/labs/lab3/src/main/java/dev/lysmux/lab3/storage/IStoryStorage.java
package dev.lysmux.lab3.storage;

import dev.lysmux.lab3.storage.exception.StoryExistsException;
import dev.lysmux.lab3.storage.exception.StoryNotFoundException;
import dev.lysmux.lab3.story.Story;

import java.util.Map;

public interface IStoryStorage {
    void save(String title, Story story) throws StoryExistsException;
    void delete(String title) throws StoryNotFoundException;
    Story get(String title) throws StoryNotFoundException;
    Map<String, Story> getAll();
}

File: proga/1_semestr/labs/lab3/src/main/kotlin/dev/lysmux/lab3/storage/IStoryStorage.kt
package dev.lysmux.lab3.storage

import dev.lysmux.lab3.storage.exception.StoryExistsException
import dev.lysmux.lab3.storage.exception.StoryNotFoundException
import dev.lysmux.lab3.story.Story

interface IStoryStorage {
    @Throws(StoryExistsException::class)
    fun save(title: String, story: Story)

    @Throws(StoryNotFoundException::class)
    fun delete(title: String)

    @Throws(StoryNotFoundException::class)
    fun get(title: String): Story

    fun getAll(): Map<String, Story>
}